### PR TITLE
Validate ShaderModule's SPIRV

### PR DESCRIPTION
**Pull request by Kangz**
PTAL @kainino0x

### CODE REVIEW

File: src/tests/unittests/validation/ShaderModuleValidationTests.cpp [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+           %main = OpFunction %void None %3
+              %5 = OpLabel
+                   OpStore %fragColor %12
+                   OpReturn
+                   OpFunctionEnd)";
+
+    utils::CreateShaderModuleFromASM(device, shader);
+}
+
+// Test case with a shader with OpUndef to test WebGPU-speciic validation
```
[the remark this answers is not in the document]
Comment (Kangz) [reply src/tests/unittests/validation/ShaderModuleValidationTests.cpp]: Done

File: src/tests/unittests/validation/ShaderModuleValidationTests.cpp [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+             %12 = OpConstantComposite %v4float %float_1 %float_0 %float_0 %float_1
+           %main = OpFunction %void None %3
+              %5 = OpLabel
+              %6 = OpUndef %v4float
+                   OpStore %fragColor %12
+                   OpReturn
+                   OpFunctionEnd)";
+
+    // Notice "%6 = OpUndef %v4float" above
+    ASSERT_DEVICE_ERROR(utils::CreateShaderModuleFromASM(device, shader));
```
[the remark this answers is not in the document]
Comment (Kangz) [reply src/tests/unittests/validation/ShaderModuleValidationTests.cpp]: Done. Added a way to get the last device error message and checked it contains the text "OpUndef".

File: src/utils/DawnHelpers.cpp [hunk 3]
```diff
@@ -25,46 +25,53 @@
 
 namespace utils {
 
+    namespace {
+
+        shaderc_shader_kind ShaderCShaderKind(dawn::ShaderStage stage) {
```
Comment (kainino0x) [inline src/utils/DawnHelpers.cpp]: nit: seems to be capitalized as Shaderc rather than ShaderC

Comment (Kangz) [reply src/utils/DawnHelpers.cpp]: Done

**Comment by Kangz** [comment]
Thanks for the reviews!

File: BUILD.gn [hunk 4]
```diff
@@ -314,6 +314,7 @@ source_set("libdawn_native_sources") {
   deps = [
     ":dawn_common",
     ":libdawn_native_utils_gen",
+    "${dawn_spirv_tools_dir}:spvtools_val",
```
Comment (SenorBlanco) [inline BUILD.gn]: Probably obvious, but I suppose this means that all Dawn clients (e.g., skia-dawn) now must have spirv-tools as a dependency?

Comment (Kangz) [reply BUILD.gn]: Yes, but it would have been the case before too if you use our BUILD.gn files.

**Pull request merged**